3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/pii/PiiPseudonymizerEnrichment.scala

```diff
@@ -0,0 +1,311 @@
+ */
+package com.snowplowanalytics
+package snowplow.enrich
+package common.enrichments.registry
+package pii
+
+// Scala
+import scala.collection.JavaConverters._
+import scala.collection.mutable.MutableList
+
+// Scala libraries
+import org.json4s
+import org.json4s.{DefaultFormats, JValue}
+import org.json4s.JsonAST._
+import org.json4s.jackson.JsonMethods
+import org.json4s.jackson.JsonMethods.{compact, parse, render}
+import org.json4s.jackson.Serialization.write
+
+// Java
+import org.apache.commons.codec.digest.DigestUtils
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClientBuilder
+import com.amazonaws.services.simplesystemsmanagement.model.GetParameterRequest
+
+// Java libraries
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode, TextNode}
+import com.jayway.jsonpath.{Configuration, JsonPath => JJsonPath}
+import com.jayway.jsonpath.MapFunction
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Iglu
+import iglu.client.validation.ProcessingMessageMethods._
+import iglu.client.{SchemaCriterion, SchemaKey}
+
+// This project
+import common.ValidatedNelMessage
+import common.utils.ScalazJson4sUtils.{extract, fieldExists}
+import common.outputs.EnrichedEvent
+
+/**
+ * Companion object. Lets us create a PiiPseudonymizerEnrichment
+ * from a JValue.
+ */
+object PiiPseudonymizerEnrichment extends ParseableEnrichment {
+
+  implicit val formats = DefaultFormats + new PiiStrategyPseudonymizeSerializer + new PiiStrategyPseudonymizeSaltSerializer
+
+  override val supportedSchema =
+    SchemaCriterion("com.snowplowanalytics.snowplow.enrichments", "pii_enrichment_config", "jsonschema", 2, 0, 0)
+
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[PiiPseudonymizerEnrichment] = {
+    for {
+      conf <- matchesSchema(config, schemaKey)
+      emitIdentificationEvent = extract[Boolean](conf, "emitIdentificationEvent").toOption
+        .getOrElse(false)
+      piiFields    <- extract[List[JObject]](conf, "parameters", "pii").leftMap(_.getMessage)
+      piiStrategy  <- extractStrategy(config)
+      piiFieldList <- extractFields(piiFields)
+    } yield PiiPseudonymizerEnrichment(piiFieldList, emitIdentificationEvent, piiStrategy)
+  }.leftMap(_.toProcessingMessageNel)
+
+  private[pii] def getHashFunction(strategyFunction: String): Validation[String, DigestFunction] =
+    strategyFunction match {
+      case "MD2"     => { DigestUtils.md2Hex(_: Array[Byte]) }.success
+      case "MD5"     => { DigestUtils.md5Hex(_: Array[Byte]) }.success
+      case "SHA-1"   => { DigestUtils.sha1Hex(_: Array[Byte]) }.success
+      case "SHA-256" => { DigestUtils.sha256Hex(_: Array[Byte]) }.success
+      case "SHA-384" => { DigestUtils.sha384Hex(_: Array[Byte]) }.success
+      case "SHA-512" => { DigestUtils.sha512Hex(_: Array[Byte]) }.success
```

Comment:
can't we just have those lines as: `DigestUtils.sha512Hex(_: Array[Byte]).success`?

Reply:
No that doesn't compile